ksp_plugin/geometric_potential_plotter.cpp

```diff
@@ -0,0 +1,213 @@
+#include "geometric_potential_plotter.hpp"
+
+#include <algorithm>
+#include <functional>
+#include <vector>
+
+#include "numerics/global_optimization.hpp"
+#include "physics/rotating_pulsating_reference_frame.hpp"
+
+namespace principia {
+namespace ksp_plugin {
+namespace _geometric_potential_plotter {
+namespace internal {
+
+using namespace principia::geometry::_barycentre_calculator;
+using namespace principia::geometry::_grassmann;
+using namespace principia::geometry::_plane;
+using namespace principia::geometry::_space;
+using namespace principia::integrators::_embedded_explicit_runge_kutta_integrator;  // NOLINT
+using namespace principia::integrators::_methods;
+using namespace principia::numerics::_global_optimization;
+using namespace principia::numerics::_root_finders;
+using namespace principia::physics::_rotating_pulsating_reference_frame;
+using namespace principia::quantities::_named_quantities;
+using namespace principia::quantities::_quantities;
+using namespace principia::quantities::_si;
+
+GeometricPotentialPlotter::GeometricPotentialPlotter(
+    not_null<Ephemeris<Barycentric>*> ephemeris)
+    : ephemeris_(ephemeris) {}
+
+void GeometricPotentialPlotter::Interrupt() {
+  plotter_ = jthread();
+  // We are single-threaded here, no need to lock.
+  plotter_idle_ = true;
+}
+
+void GeometricPotentialPlotter::RequestEquipotentials(
+    Parameters const& parameters) {
+  last_parameters_ = parameters;
+  absl::MutexLock l(&lock_);
+  // Only process this request if there is no analysis in progress.
+  if (plotter_idle_) {
+    plotter_idle_ = false;
+    plotter_ = MakeStoppableThread(
+        [this, parameters]() { PlotEquipotentials(parameters).IgnoreError(); });
```

Comment:
We generally don't write bodies on a single line, so line breaks after `{` and before `}`.

Reply:
We do for λs.
https://github.com/mockingbirdnest/Principia/blob/3fcf46cdbff6b8a51c65bb73e1a3930f7e2880a4/base/recurring_thread_body.hpp#L29